src/components/select/editForm/Select.edit.data.js

```diff
@@ -434,6 +454,18 @@ export default [
       },
     },
   },
+  {
+    key: 'data.async',
```

Comment:
Why does this need to be an option?... Why can't we just look at the object that is returned from the execution and if a Promise is returned, then it becomes "async".